xcode/Subconscious/Shared/Models/Slashlink.swift

```diff
@@ -16,7 +16,7 @@ public struct Slashlink:
     Codable,
     LosslessStringConvertible
 {
-    private static let slashlinkRegex = /(\@(?<petname>[\w\d\-]+))?\/(?<slug>(?:[\w\d\-]+)(?:\/[\w\d\-]+)*)/
+    private static let slashlinkRegex = /(\@(?<petname>[\w\d\-\.]+))?(\/(?<slug>(?:[\w\d\-]+)(?:\/[\w\d\-]+)*))?/
```

Comment:
1. Allow .'s in petname
2. Allow for missing slug component (including the `/`)

Reply:
I believe this should be:

```regex
(\@(?<petname>(?:[\w\d\-]+)(?:\.[\w\d\-]+)*))?(\/(?<slug>(?:[\w\d\-]+)(?:\/[\w\d\-]+)*))?
```

(Carrying over the same petname grammar from `Petname`)

Reply:
We should write some tests covering typical and pathological parsing cases. Off the top of my head.

```swift
XCTAssertNotNil(Slashlink("@bob"))
XCTAssertNotNil(Slashlink("@alice.bob"))
XCTAssertNotNil(Slashlink("@dan.charlie.bob.alice"))
XCTAssertNotNil(Slashlink("@bob-foo.alice-foo"))
XCTAssertNotNil(Slashlink("@bob_foo.alice_foo"))
XCTAssertNil(Slashlink("@eve...alice"))
XCTAssertNil(Slashlink("@.eve.alice"))
XCTAssertNil(Slashlink("@alice.eve."))
XCTAssertNil(Slashlink("@@eve.alice"))
XCTAssertNil(Slashlink("@eve@alice"))
XCTAssertNil(Slashlink("@eve//foo"))
XCTAssertNil(Slashlink("@eve..alice/foo"))
XCTAssertNil(Slashlink("@eve.evelyn/foo//bar"))
```

Reply:
Added these cases and updated the regex